comm.go

```diff
@@ -156,16 +158,20 @@ func (p *PubSub) handlePeerEOF(ctx context.Context, s network.Stream) {
 }
 
 func (p *PubSub) handleSendingMessages(ctx context.Context, s network.Stream, outgoing <-chan *RPC) {
-	bufw := bufio.NewWriter(s)
-	wc := protoio.NewDelimitedWriter(bufw)
+	writeRpc := func(rpc *RPC) error {
+		size := uint64(rpc.Size())
 
-	writeMsg := func(msg proto.Message) error {
-		err := wc.WriteMsg(msg)
+		buf := pool.Get(varint.UvarintSize(size) + int(size))
+		defer pool.Put(buf)
+
+		n := binary.PutUvarint(buf, size)
+		_, err := rpc.MarshalTo(buf[n:])
 		if err != nil {
 			return err
 		}
 
-		return bufw.Flush()
+		_, err = s.Write(buf)
+		return err
```

Comment:
Notes for @vyzo
* I assume the bufio was used as protois's writer could do two writes, and we wanted to avoid that.
	* Now we do only one write so I removed the buffer
* I decided not to reuse the msgio Writer as that would require getting buffer from the pool twice
	* To marshal proto first
	* And then inside the Writer's `WriteMsg`
	* With this, less abstracted way, we do it only once
* The writeRpc logic is mostly copied from  https://github.com/celestiaorg/go-libp2p-messenger. It's [serde](https://github.com/celestiaorg/go-libp2p-messenger/blob/main/serde/serde.go#L74) pkg provides similar functionality to msgio, but it abstracts the message, not the Reader. Subsequently, it doesn't have an issue from the point above and provides a simpler API, IMO. It uses the go-buffer-pool. Just sharing.

Reply:
yup, makes sense.